src/entities/migratePool/liquidityBootstrapping/index.ts

```diff
@@ -0,0 +1,124 @@
+    MigratePoolLiquidityBootstrappingInput,
+    MigratePoolLiquidityBootstrappingQueryInput,
+    MigratePoolLiquidityBootstrappingQueryOutput,
+} from '../types';
+import { balancerMigrationRouterAbiExtended } from '@/abi';
+import { getRandomBytes32 } from '@/entities/utils/getRandomBytes32';
+import { Permit2 } from '@/index';
+
+export class MigratePoolLiquidityBootstrapping implements MigratePoolBase {
+    public async query(
+        input: MigratePoolLiquidityBootstrappingQueryInput,
+        block?: bigint,
+    ): Promise<MigratePoolLiquidityBootstrappingQueryOutput> {
+        const client = createPublicClient({
+            transport: http(input.rpcUrl),
+            chain: CHAINS[input.chainid],
+        });
+
+        const args = [
+            input.pool,
+            input.sender,
+            input.excessReceiver,
+            {
+                name:
+                    input.weightedPoolParams.name ||
+                    input.weightedPoolParams.symbol,
+                symbol: input.weightedPoolParams.symbol,
+                roleAccounts: {
+                    pauseManager: input.weightedPoolParams.pauseManager,
+                    swapFeeManager: input.weightedPoolParams.swapFeeManager,
+                    poolCreator:
+                        input.weightedPoolParams.poolCreator ?? zeroAddress,
+                },
+                swapFeePercentage: input.weightedPoolParams.swapFeePercentage,
+                poolHooksContract: input.weightedPoolParams.poolHooksContract,
+                enableDonation: input.weightedPoolParams.enableDonation,
+                disableUnbalancedLiquidity:
+                    input.weightedPoolParams.disableUnbalancedLiquidity,
+                salt: input.weightedPoolParams.salt || getRandomBytes32(),
+            },
+        ] as const;
+
+        const { result } = await client.simulateContract({
+            address: AddressProvider.LBPoolMigrationRouter(input.chainid),
+            abi: balancerMigrationRouterAbiExtended,
+            functionName: 'queryMigrateLiquidity',
+            args,
+            blockNumber: block,
+        });
+        const [exactAmountsIn, bptAmountOut] = result;
+
+        return {
+            bptAmountOut: bptAmountOut,
+            exactAmountsIn: exactAmountsIn,
+            poolType: 'LiquidityBootstrapping',
+            pool: input.pool,
+            chainId: input.chainid,
+            to: AddressProvider.LBPoolMigrationRouter(input.chainid),
+        };
+    }
+
+    public buildCall(
+        input: MigratePoolLiquidityBootstrappingInput,
+    ): MigratePoolBuildCallOutput {
+        const callData = this.encodeCall(input);
+        return {
+            callData,
+            to: AddressProvider.LBPoolMigrationRouter(input.chainid),
+        };
+    }
+    public buildCallWithPermit2(
+        _input: MigratePoolLiquidityBootstrappingInput,
+        _permit2: Permit2,
+    ): MigratePoolBuildCallOutput {
+        // not implemented
```

Comment:
Any reason why this wasn't implemented? If the idea is to avoid blocking this PR from being released, I'd say it would be nice to create a ticket on the project board to handle this in the future.

Reply:
I have now removed that functionality from the `MigratePoolBase` interface. The Migration Router does not support Permit in its current iteration.